src/esp/gfx/Renderer.cpp

```diff
@@ -164,7 +165,9 @@ struct Renderer::Impl {
       } else if (type == sensor::SensorType::Semantic) {
         shader->bindObjectIdTexture(tgt.getObjectIdTexture());
       }
-      shader->setColorMapTransformation(colorMapOffset, colorMapScale);
+      if ((colorMapOffset >= 0) && (colorMapScale >= 0)) {
+        shader->setColorMapTransformation(colorMapOffset, colorMapScale);
```

Comment:
As we discussed :) the semantic visualization and the depth visualization each use different instances of TextureVisualizerShader and different compiled frag shader programs.  Because of this (that both use cases have their own ShaderProgram and compiled frag shader) they do not "cross-pollinate" their values, so it is reasonable to assume that these values will not need to be reset on subsequent draws.